swarm/src/behaviour.rs

```diff
@@ -187,61 +205,99 @@ pub trait NetworkBehaviour: 'static {
     ) {
     }
 
-    /// Informs the behaviour about an event generated by the handler dedicated to the peer identified by `peer_id`.
-    /// for the behaviour.
-    ///
-    /// The `peer_id` is guaranteed to be in a connected state. In other words,
-    /// [`NetworkBehaviour::inject_connection_established`] has previously been called with this `PeerId`.
-    fn inject_event(
-        &mut self,
-        peer_id: PeerId,
-        connection: ConnectionId,
-        event: <<Self::ConnectionHandler as IntoConnectionHandler>::Handler as ConnectionHandler>::OutEvent,
-    );
-
-    /// Indicates to the behaviour that the dial to a known or unknown node failed.
-    fn inject_dial_failure(
-        &mut self,
-        _peer_id: Option<PeerId>,
-        _handler: Self::ConnectionHandler,
-        _error: &DialError,
-    ) {
-    }
-
-    /// Indicates to the behaviour that an error happened on an incoming connection during its
-    /// initial handshake.
-    ///
-    /// This can include, for example, an error during the handshake of the encryption layer, or the
-    /// connection unexpectedly closed.
-    fn inject_listen_failure(
-        &mut self,
-        _local_addr: &Multiaddr,
-        _send_back_addr: &Multiaddr,
-        _handler: Self::ConnectionHandler,
-    ) {
-    }
+    // TODO: Can not deprecate as it takes ownership of handler event.
```

Comment:
We can deprecate them + make a default implementation where we call `on_event`. Thus, the moment users migrate, they should receive the event via `on_event`.

Reply:
:+1: wonderful. Done with https://github.com/libp2p/rust-libp2p/pull/2867/commits/de5c46dbe198c756c27de467196324084dee86b8.